src/engine/Server.cpp

```diff
@@ -431,9 +431,11 @@ CPP_template_2(typename RequestT, typename ResponseT)(
     }
   }
 
+  // Store the QueryExecutionTree in case of errors.
```

Comment:
```suggestion
  // Store the QueryExecutionTree outside of the lambda, s.t. we have access in case or errors to create an informative error message that includes the runtime information.
```